include/openPMD/Datatype.hpp

```diff
@@ -177,10 +189,338 @@ determineDatatype(std::shared_ptr< T >)
 
     return DT::UNDEFINED;
 }
-} // openPMD
+
+/** Return number of bytes representing a Datatype
+ *
+ * @param d Datatype
+ * @return number of bytes
+ */
+inline size_t
+toBytes( Datatype d )
+{
+    using DT = Datatype;
+    switch( d )
+    {
+        case DT::CHAR:
+        case DT::VEC_CHAR:
+        case DT::STRING:
+        case DT::VEC_STRING:
+            return sizeof(char);
+            break;
+        case DT::UCHAR:
+        case DT::VEC_UCHAR:
+            return sizeof(unsigned char);
+            break;
+        // case DT::SCHAR:
+        // case DT::VEC_SCHAR:
+        //     return sizeof(signed char);
+        //     break;
+        case DT::SHORT:
+        case DT::VEC_SHORT:
+            return sizeof(short);
+            break;
+        case DT::INT:
+        case DT::VEC_INT:
+            return sizeof(int);
+            break;
+        case DT::LONG:
+        case DT::VEC_LONG:
+            return sizeof(long);
+            break;
+        case DT::LONGLONG:
+        case DT::VEC_LONGLONG:
+            return sizeof(long long);
+            break;
+        case DT::USHORT:
+        case DT::VEC_USHORT:
+            return sizeof(unsigned short);
+            break;
+        case DT::UINT:
+        case DT::VEC_UINT:
+            return sizeof(unsigned int);
+            break;
+        case DT::ULONG:
+        case DT::VEC_ULONG:
+            return sizeof(unsigned long);
+            break;
+        case DT::ULONGLONG:
+        case DT::VEC_ULONGLONG:
+            return sizeof(unsigned long long);
+            break;
+        case DT::FLOAT:
+        case DT::VEC_FLOAT:
+            return sizeof(float);
+            break;
+        case DT::DOUBLE:
+        case DT::VEC_DOUBLE:
+        case DT::ARR_DBL_7:
+            return sizeof(double);
+            break;
+        case DT::LONG_DOUBLE:
+        case DT::VEC_LONG_DOUBLE:
+            return sizeof(long double);
+            break;
+        case DT::BOOL:
+            return sizeof(bool);
+            break;
+        case DT::DATATYPE:
+        case DT::UNDEFINED:
+            throw std::runtime_error("toBytes: Invalid datatype!");
+            break;
+    }
+    return 0;
+}
+
+/** Return number of bits representing a Datatype
+ *
+ * @param d Datatype
+ * @return number of bits
+ */
+inline size_t
+toBits( Datatype d )
+{
+    return toBytes( d ) * CHAR_BIT;
```

Comment:
Just out of curiosity: Are you aware of any platform where `CHAR_BIT != 8`?

Reply:
I just like being explicit. Of course, if I should ever get my hands on a C++11 capable compiler running on a UNIVAC ... <3 Besides that, some DSPs like TexasInstrucments are using 16 and 32bit Bytes.
Found this SO on it: https://stackoverflow.com/questions/5516044/system-where-1-byte-8-bit